### ISSUE TITLE
[BUG] ReactiveUI navigation doesn't change the content of the page

### ISSUE THREAD

**Issue by vladislav-itm** [reactiveui/ReactiveUI#3035]
**Describe the bug**
Whe using ReactiveUI nagivation (using IScreen, RoutingState, IRoutableViewModel, etc.), the page content doesn't change when navigating to another page

**Steps To Reproduce**
A repo with a complete MAUI app showing the problem
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue

Steps to reproduce:
1. Create MAUI project from default VS 2022 template (VS 17.1.0 Preview 1.1)
2. Install ReactiveUI.Maui 16.3.10 package
3. Create view models for main page and another page (to which we will navigate). Both implement IRoutableViewModel
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/MainPageViewModel.cs
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/AnotherViewModel.cs
4. Modify MainPage.xaml and MainPage.xaml.cs to use ReactiveContentPage, remove unused controls. Bind button to view model's navigation command
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/MainPage.xaml
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/MainPage.xaml.cs
5. Create "AnotherPage" and its code-behind file which use ReactiveContentPage
code
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/AnotherPage.xaml
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/AnotherPage.xaml.cs
6. Create AppBootstrapper which will host "RoutingState", register views to DI container and return main page as "RoutedViewHost"
code
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/AppBootstrapper.cs
7. Modify App.xaml.cs to instantiate AppBootstrapper and create main page
https://github.com/vladislav-itm/ReactiveUI-Navigation-Issue/blob/master/MauiApp5/App.xaml.cs
8. Run the application
9. Click "Click me" button
10. Title of the page specified in "UrlPathSegment" gets changed to new one
11. But the page content does change to new one

**Expected behaviour**
Page content gets changed after navigating to another view model (page)

**Screenshots**
Main page before button click:
![image](https://user-images.githubusercontent.com/53902622/143408078-7c2af0fe-033d-4907-8a95-1dfde9f17a03.png)

Page after button click (content doesn't change):
![image](https://user-images.githubusercontent.com/53902622/143408135-fbff5a4d-4049-426f-8402-312b5aa43ede.png)

**Environment**
- OS:  Windows 10
- Device:  Android emulator: Pixel 2 - API 28 (Android 9.0 - API 28)
- ReactiveUI Version:    16.3.10 

**Additional context**
I noticed that I am not the only one experiencing this problem. Other devs in ReactiveUI Slack told about this aswell

**Comment by vladislav-itm** [comment]
Jesus, I thought the code in issue will look as expected... Sorry about that! I attached a repo

**Comment by aschaefer94** [comment]
I ran into this same issue when trying out the latest MAUI preview.

**Comment by glennawatson** [comment]
Yeah our Maui experience will be raw for a while just due to its delayed release, we'll be getting it more streamlined before the main Maui release. Keep the issues coming so we can get it solid before then.

**Comment by ili** [comment]
I'v updated original sample to the latest MAUI & ReactiveUI (https://github.com/ili/ReactiveUI-Navigation-Issue), next problems a there with navigation:

1) On application start when main page is loaded Navigation is not empty - we can go back to the same `MainViewModel` (`MainView`)
2) If we do click `Back` burron and then click `Navigate 1` button we'll be navigated to the `AnotherViewModel` (`AbotherView`), BUT: `Back` button disappears (as like for `NavigateAndReset` but `Navigate` is used)
3) `Navigate 2` button - I'm trying to navigate with "history", but nothing happens: 
```
                HostScreen.Router.NavigationStack.Clear();
                HostScreen.Router.NavigationStack.AddRange(new IRoutableViewModel[]
                {
                    new AnotherViewModel(),
                    new AnotherViewModel2()
                });
```
4) `Navigate 3` - mostly the same, navigation occures, but again without back button:
```
                HostScreen.Router.NavigationStack.Clear();
                HostScreen.Router.NavigationStack.Add(new AnotherViewModel());

                HostScreen.Router.Navigate.Execute(new AnotherViewModel2()).Subscribe();
```

**Comment by ChrisPulman** [comment]
Fixed by #3303
